jdbc/src/main/java/com/interface21/jdbc/core/JdbcTemplate.java

```diff
@@ -69,32 +69,17 @@ public <T> T execute(String sql, SqlFunction<PreparedStatement, T> action) {
             log.error("Error executing query: {}", e.getMessage(), e);
             throw new JdbcAccessException("Error executing query: " + sql, e);
         } finally {
-            closeConnection(connection);
+            closeConnection(isNewConnection, connection);
```

Comment:
```suggestion
            closeIfNecessary(isNewConnection, connection);
```

항상 close하는 것이 아닌 분기처리하고 특정 상황에서만 close하는 메서드네요.
이름 이렇게 하는건 어떨까요? 단순 제안이라 넘어가도 돼요!

Reply:
isNewConnection을 내부에서 연산할 수 있는 메서드면 closeConnection이 더 자연스러울 것 같은데, 외부에서 isNewConnection에 대한 정보를 받아와야 하니 특정 상황에서만 자원을 닫는 걸 명시해도 좋을 것 같네요 👍